src/main.rs

```diff
@@ -1,6 +1,9 @@
+use std::error::Error;
+
 mod cli;
 mod mrdog;
 
-fn main() {
-    cli::run();
+fn main() -> Result<(), Box<dyn Error>> {
+    cli::run()?;
+    Ok(())
```

Comment:
NIT: Typically, the main fn does not have an error result (nobody process it anyway), so it returns void and panics. You can achieve this by `run().unwrap()` I think.

Reply:
I see, I thought I saw somewhere the `-> Result<>`. Here it seems like an option https://doc.rust-lang.org/rust-by-example/error/result.html?highlight=main#using-result-in-main. Seems like it should print out the error using the `Debug` trait (though no clue what happens for this case I made here 😅)